src/backend/backend_concept.h

```diff
@@ -15,20 +17,29 @@ namespace snmalloc
    * get_metadata takes a bool-ean template parameter indicating whether it may
    * be accessing memory that is not known to be committed.
    */
-  template<typename Meta>
+  template<
+    typename Meta,
+    typename BackendMetadata = std::remove_pointer_t<
+      decltype(std::declval<Meta>()
+                 .template get_metaentry_mut<true>(std::declval<address_t>())
+                 .get_meta())>>
   concept ConceptBackendMeta =
-    requires(
-      address_t addr,
-      size_t sz,
-      const MetaEntry& t)
+    requires(address_t addr, size_t sz, const MetaEntry<BackendMetadata>& t)
   {
-    { Meta::set_metaentry(addr, sz, t) } -> ConceptSame<void>;
+    {
+      Meta::set_metaentry(addr, sz, t)
+    }
+    ->ConceptSame<void>;
 
-    { Meta::template get_metaentry<true>(addr) }
-      -> ConceptSame<const MetaEntry&>;
+    {
+      Meta::template get_metaentry<true>(addr)
+    }
+    ->ConceptSame<const MetaEntry<BackendMetadata>&>;
```

Comment:
This does not return a `MetaEntry<BackendMetadata>&`, but a `PageMapEntry<BackendMetadata>&`.

Reply:
@nwf, can you help with the concepts goo?

Reply:
I believe this is fixed now.

Reply:
Yes, though it looks like you let clangformat get its hands on the concept syntax.  I don't really object to its layout of compound `require`ments, even if it's not how I have been writing them, but the asymmetry of `foo&& bar` is unfortunate.

Reply:
Sorry, I forgot clang-format didn't do a good job here.